core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -39,12 +38,32 @@ import org.apache.zookeeper.AsyncCallback.{DataCallback, StringCallback}
 import org.apache.zookeeper.KeeperException.Code
 import org.apache.zookeeper.data.{ACL, Stat}
 import org.apache.zookeeper.{CreateMode, KeeperException, ZooDefs, ZooKeeper}
+import kafka.utils.Json._
+import org.apache.zookeeper.Watcher.Event.KeeperState
+
 
 import scala.collection._
 import scala.collection.JavaConverters._
 
+// This is copy of the KeeperState with addition
+// of SessionEstablishmentError and NewSession
+object ZKState extends Enumeration{
+  type String = Value;
+  val Unknown = Value("Unknown")
+  val SessionEstablishmentError = Value("SessionEstablishmentError")
+  val NewSession = Value("NewSession")
+  val Disconnected = Value("Disconnected")
+  val SyncConnected = Value("SyncConnected")
+  val AuthFailed = Value("AuthFailed")
+  val ConnectedReadOnly = Value("ConnectedReadOnly")
+  val SaslAuthenticated = Value("SaslAuthenticated")
+  val Expired = Value("Expired")
+}
```

Comment:
We probably want to track these metrics in ZookeeperClient since that's where the ZK states are propagated. However, if you don't have the time, you can just leave this part out and we can address this in a separate jira.

Reply:
Separate Jira sounds better to me since we already have one for the ZKClient metrics